### PR TITLE
Dat 1262 beamer implementation

### PR DESCRIPTION
**Pull request by jcamposmk**
### **Add Beamer**
**Jira Ticket:** https://makingsense.atlassian.net/browse/DAT-1262
⚠️ ⚠️**Note:** Work is being done on customizing the tool to adjust colors and categories of notifications
 
**RESULT:**
Grabación de pantalla 2022-12-07 a la(s) 7 37 51 a

### CODE REVIEW

**Comment by andresmoschini** [comment]
Sorry @jcamposmk, I forgot to review it. I am going to review it in the morning.

File: src/components/Notifications.tsx [hunk 1]
```diff
@@ -1,56 +1,47 @@
-import { useMemo, useState } from "react";
-import useOnclickOutside from "react-cool-onclickoutside";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (andresmoschini) [inline src/components/Notifications.tsx:2]: Are these patches really necessary?

Comment (jcamposmk) [reply src/components/Notifications.tsx:2]: me parece que fue porque daba error al importar Beamer
Captura de Pantalla 2022-12-19 a la(s) 8 42 45 a  m

sabes si hay una mejor forma de hacerlo?

Comment (andresmoschini) [reply src/components/Notifications.tsx:2]: Did you try adding a reference to `@types/react-beamer`? does this library exist?

Comment (andresmoschini) [reply src/components/Notifications.tsx:2]: If the typings do not exist for this library, you can declare the type in the project. It is explained in the first part of [this article](https://pjausovec.medium.com/how-to-fix-error-ts7016-could-not-find-a-declaration-file-for-module-xyz-has-an-any-type-ecab588800a8).

The advantage of doing this is that we can improve the typings when needed.

Comment (jcamposmk) [reply src/components/Notifications.tsx:2]: Yes I tried, but give me an error: Not found

File: src/components/Notifications.tsx [hunk 2]
```diff
@@ -1,56 +1,47 @@
...
+import { User } from "../model";
+import { defaultLanguage } from "./i18n/MenuIntlProvider";
+import { useAppConfiguration } from "../AppConfiguration";
 
-interface NotificationsProps {
-  notifications: ReadonlyArray<string>;
-  emptyNotificationText: string;
-}
+// this is a temporary beamer account with limited functions
+export const DEFAULT_BEAMER_ID = "wCWRrvwa50706";
```
Comment (andresmoschini) [inline src/components/Notifications.tsx:10]: This value should not be here.

Alternatives could be:
* Adding a default configuration file [like this](https://github.com/FromDoppler/doppler-editors-webapp/blob/main/src/default-configuration.ts)
* Setting this in the configuration of your local environment [like here](https://github.com/FromDoppler/doppler-menu-mfe/blob/main/public/index.html#L30)

File: src/components/Notifications.tsx [hunk 3]
```diff
@@ -1,56 +1,47 @@
...
-  emptyNotificationText,
-}: NotificationsProps) => {
-  const count = notifications.length;
-  const [openNotification, setOpenNotification] = useState(false);
-  const notificationsRef = useOnclickOutside(() => setOpenNotification(false));
+export const BEAMER_CONFIG = {
+  selector: ".beamer-icon",
+  top: -10,
+  right: -10,
+  language: defaultLanguage,
```
Comment (andresmoschini) [inline src/components/Notifications.tsx:17]: You should use the selected language, right?

File: src/components/i18n/MenuIntlProvider.tsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { messages_es } from "./es";
 import messages_en from "./en";
 import { flattenMessages } from "./utils";
 
-const defaultLanguage = "es";
+export const defaultLanguage = "es";
```
Comment (andresmoschini) [inline src/components/i18n/MenuIntlProvider.tsx:8]: I would revert this change.

File: src/components/Notifications.tsx (hunk 3 above)
Comment (jcamposmk) [reply src/components/Notifications.tsx:17]: yes, it's right ... I use it to initialize beamer ... in this part: 
```
 beamerInitialize(beamerId, BEAMER_URL, {
      ...BEAMER_CONFIG,
      //---------------Visitor Information---------------
      user_firstname: user.fullname,
      user_email: user.email,
      language: user.lang,
    });
```

File: src/components/Notifications.tsx (hunk 1 above)
Comment (andresmoschini) [reply src/components/Notifications.tsx:2]: > Yes I tried, but give me an error: Not found

You are right because there are no defined types for `react-beamer`: https://github.com/search?q=repo%3ADefinitelyTyped%2FDefinitelyTyped+%22react-beamer%22&type=code

Maybe the other solution could work (https://github.com/FromDoppler/doppler-menu-mfe/pull/234#discussion_r1052221998), in my opinion, it is slightly better, but if it does not work, the patch is ok.

File: src/components/Notifications.tsx [hunk 5]
```diff
@@ -1,56 +1,47 @@
...
+import { defaultLanguage } from "./i18n/MenuIntlProvider";
+import { useAppConfiguration } from "../AppConfiguration";
 
-interface NotificationsProps {
-  notifications: ReadonlyArray<string>;
-  emptyNotificationText: string;
-}
+// this is a temporary beamer account with limited functions
+export const DEFAULT_BEAMER_ID = "wCWRrvwa50706";
+export const BEAMER_URL = null;
```
Comment (andresmoschini) [inline src/components/Notifications.tsx:11]: We do not need it in another place, right?

```suggestion
const BEAMER_URL = null;
```

File: src/components/Notifications.tsx [hunk 6]
```diff
@@ -1,56 +1,47 @@
...
 
-export const Notifications = ({
-  notifications,
-  emptyNotificationText,
-}: NotificationsProps) => {
-  const count = notifications.length;
-  const [openNotification, setOpenNotification] = useState(false);
-  const notificationsRef = useOnclickOutside(() => setOpenNotification(false));
+export const BEAMER_CONFIG = {
```
Comment (andresmoschini) [inline src/components/Notifications.tsx:13]: We do not need it in another place, right?

```suggestion
const BEAMER_CONFIG = {
```

File: src/components/i18n/MenuIntlProvider.tsx (hunk 4 above)
Comment (jcamposmk) [reply src/components/i18n/MenuIntlProvider.tsx:8]: you say it for this: `Adding a default configuration file` [like this](https://github.com/FromDoppler/doppler-editors-webapp/blob/main/src/default-configuration.ts)

Comment (andresmoschini) [reply src/components/i18n/MenuIntlProvider.tsx:8]: I think that default configuration is not related with this line.

`defaultLanguage` should only be used inside `MenuIntlProvider.tsx`, so, you do not need to export it.

File: src/components/Notifications.tsx (hunk 2 above)
Comment (jcamposmk) [reply src/components/Notifications.tsx:10]: in all place where the menu MFE is used, I give it of beamerId [like this](https://github.com/FromDoppler/doppler-editors-webapp/blob/main/src/default-configuration.ts). Do you think I can remove this line that you are telling me about?

Comment (andresmoschini) [reply src/components/Notifications.tsx:10]: I think that this line can be safely removed because you will have the ID from the default configuration. Am I missing something?

Comment (jcamposmk) [reply src/components/Notifications.tsx:10]: it's correct!
here, I give the ID in the Webapp: https://github.com/FromDoppler/doppler-webapp/pull/2199/commits/d9ef288af7dabd7f872e538fd3338279e75eae2c

in Doppler: https://github.com/MakingSense/Doppler/pull/9891/commits/275bb21382123371cf6bf3691f3ac4d2b5a2375d
